### PR TITLE
Feature/flip card as

### PR DESCRIPTION
**Pull request by AsherSpurr**
# Pull Request
Feature/flip-card-as
## Description
Create flip card and fetch for each tea
## Screenshots (if applicable)
Screenshot 2024-04-08 at 6 48 35 PM

## Changes Made
Addditions:
- Finish fetch for singular tea
- async function fetchTea in Card.tsx -> awaits tea data 
- targetTea state -> returned data from fetchTEa
- Variables CSS && ncss (no CSS) -> This was tricky -- CSS wouldn't transform the cards back these variables were a way to keep the card flipping
- Adds isClicked state which is set to a boolean -> This allows for the CSS to be applied or not
- Needed to apply this same logic to the text because the default transform 180degrees was not being applied in CSS
- Adjust a LOT of styling to the cards -> adding this flip feature wrecked the grid layout at first -> it is now resolved
- Move the favorites button outside of the card container so it doesn't flip (Could not move the text to do the same without a lot of re-styling)
## Checklist
- [x] The code follows the project's coding standards.
- [ ] Unit tests have been added or updated to cover the changes.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] The code compiles without errors.
- [ ] The changes have been tested locally and pass all relevant tests.
- [ ] All new and existing tests pass.
- [x] The pull request has been reviewed by at least one other contributor.

### CODE REVIEW

**Review by corysanders3 — COMMENTED** [review]
Dove into the code updates. Looks good. Really great css styling for this. There are some merge conflicts, so reviewing those next.

**Review by corysanders3 — APPROVED** [review]
Updated merge conflicts. Looks good from my end, but if reviewer 2 can look over the files changed, that would be great. Thanks!

File: src/components/card/Card.tsx [hunk 1]
```diff
@@ -12,34 +13,68 @@ interface CardProp {
...
     addFavs(tea);
     setColor(color === "#B1AE91" ? "#895B1E" : "#B1AE91");
   }
 
+  async function fetchTea(e: React.MouseEvent<HTMLDivElement>): Promise<any> {
+    let tea = await fetchSingleTea(slug)
+    let target = tea[0]
+    setTargetTea(target)
+  }
+
```
Comment (tednaphil) [inline src/components/card/Card.tsx:42]: Do we need to add any error handling for the singular fetches on the `Card`s?

File: src/components/card/Card.tsx [hunk 2]
```diff
@@ -12,34 +13,68 @@ interface CardProp {
...
+
   function favTea(e: React.MouseEvent<HTMLButtonElement>) {
     addFavs(tea);
     setColor(color === "#B1AE91" ? "#895B1E" : "#B1AE91");
   }
 
+  async function fetchTea(e: React.MouseEvent<HTMLDivElement>): Promise<any> {
+    let tea = await fetchSingleTea(slug)
+    let target = tea[0]
```
Comment (tednaphil) [inline src/components/card/Card.tsx:39]: Still chuckling at the array of one solitary item . Thanks for lending your creativity!

**Review by tednaphil — APPROVED** [review]
Our favorites test was failing, so I updated the tests to be able to access the `fav-btn` elements! The animation is fabulous! Thank you @AsherSpurr

**Pull request merged**